scripts/modules/b2b-account/quotes.js

```diff
@@ -37,17 +39,74 @@ define([
         render: function () {
             var self = this;
             Backbone.MozuView.prototype.render.apply(this, arguments);
-            var collection = new QuotesGridCollectionModel({ autoload: true });
+            var collection = new QuotesGridCollectionModel({ autoload: true });            
+            $('[data-mz-action="applyfilter"]').on('change input', function(e) {
+                e.preventDefault();
+                FILTERSTRING = "";
+                var dateValue ="";
+                var nameValue = $(this).val();
+                if($("#expirationdate").val()!=="")
+                {                    
+                    dateValue  = $("#expirationdate").val();
+                }
+                self.filterGrid(nameValue, dateValue, collection);
+                
+                });
+             $('[data-mz-action="applyDatefilter"]').on('change', function(e) {
+                e.preventDefault();
+               
+                var nameValue ="";
+                if($("#searchName").val()!=="")
+                {
+                    nameValue  = $("#searchName").val()+" and ";
+                }
+
+                var dateValue =  $(this).val();                    
+                self.filterGrid(nameValue, dateValue, collection);             
+               
+                });
             this.initializeGrid(collection);
         },
 
+        filterGrid: function (nameValue, dateValue, collection)
+        {
+            FILTERSTRING = "";
+             if(nameValue!=="")
+             {
+                 nameValue = NAMESTRING + nameValue;
+                 FILTERSTRING = nameValue;
+                 if(dateValue!=="")
+                 {
+                     dateValue = DateString + dateValue+"T00:00:00z";
```

Comment:
Make this hardcoded timestamp as a 'const'

Reply:
const not supported but seperated it in a variable and used.